tests/test_uri.py

```diff
@@ -358,11 +358,16 @@ def testQuoting(self):
 class WebdavURITestCase(unittest.TestCase):
 
     def setUp(self):
+        os.environ["WEBDAV_AUTH_METHOD"] = "TOKEN"
```

Comment:
I think you should use unittest.mock.patch here to patch the environment so we know we aren't contaminating other tests that might be called later.  I haven't entirely thought about how this can be done without having the env set for each test separately.

Reply:
Done in ebf3ebe.
`@unittest.mock.patch.dict` class decorator only applies to methods with `test` prefix in the name, which means it doesn't apply to setUp() and tearDown() for example: I had to manually decorate those methods in test_butler.py.